### ISSUE TITLE
Can I avoid the use of the Compass

### ISSUE THREAD

**Issue by nalmeida** [at-import/Singularity#44]
Is it possible to run Singularity just using SASS, without Compass?

**Comment by scottkellum** [comment]
No. The math and modular-scale dependencies hook into Ruby natively through Compass. This is done to speed up compile time and calculate things more accurately.

**Comment by Snugug** [comment]
TL;DR: No. (at least not without lots of work on your part)

Singularity by design is a Compass extension, and relies on other Compass extensions to work, including some written entirely in Ruby. If you were interested in running Singularity w/o Compass, you could just use the partials, but you would need to include all of the partials of all of its dependencies, rewrite all Compass functions/mixins that we use, and get the Ruby functions working in vanilla Sass, which is doable, but by no means just a drop in solution. You'd also need to diff any updates to Singularity and it's dependencies yourself and update manually.
If you're willing to do all of this, then yes. Otherwise, no. You should use Compass, and use Singularity as a Compass extension.

**Comment by lmartins** [comment]
Hey guys, sorry to dug this issue but if at some point the Compass requirement can be avoided please consider it.
Would love to use Singularity but im now too used to Libsass (which doesn't support Compass) to go back to Ruby based compilers.
Thanks.

**Comment by laurelnaiad** [comment]
It would not be possible to whine in a way that would trump this but i thought I might try nonetheless.

**Comment by lmartins** [comment]
The use case is to be able to compile using libsass, although Chris mentioned it would probably try to make compass work with libsass too: https://github.com/gruntjs/grunt-contrib-sass/issues/38

**Comment by Snugug** [comment]
While I appreciate the want to use this without Compass, and we'd like to support that use case, it is simply not possible at the moment. When Sass 3.3 is available, we will be able to distribute Singularity without it being tied to Compass, but it will still only work with a compiler that is compatible with Sass 3.3 (libsass won't have compatibility immediately, for instance). Additionally, we won't be able to ensure dependencies are met, and as such Singularity (especially responsive grid syntax) won't necessarily work out of the box

**Comment by pepelsbey** [comment]
Another reason for you guys to untie Singularity from Compass: since Susy (alternative grid system you’re probably aware of) managed to remove Compass as a dependency [in v2.0.0](http://susydocs.oddbird.net/en/latest/changelog/#alpha-5-nov-25-2013) it became my choice. So when I tried to give Singularity a chance I stuck with all its “rubyness” and gave up. I use Susy as just a bunch of files imported in my Sass, I wish Singularity could give me the same simplicity one day.

**Comment by Snugug** [comment]
Singularity can be installed through bower without Compass. Sass 3.3 bring released was a blocker, not any longer.

`bower install singularitygs`

**Comment by pepelsbey** [comment]
Rather just `bower install singularity` without `gs`, but still good news!

But I wonder why it’s not mentioned anywhere on wiki?

**Comment by Snugug** [comment]
Docs are being updated for 1.2

**Comment by newtonianb** [comment]
Tying it to compass is a terrible move, the last 2 large scale projects I've worked on have dropped compass in favor of super fast node sass compiler with extra libraries like post-css and bourbon to replace the missing functionality. Compass is way too slow on medium to large size project, I wouldn't be surprised the project goes defunct. This should be seperated

**Comment by scottkellum** [comment]
@newtonianb Please note that Compass is no longer a requirement. There are [instructions to install with Bower and Eyeglass in the readme](https://github.com/at-import/Singularity#singularity-quickstart).

If you post any further comments with this tone I will not hesitate to delete and block them. This project and Compass are projects that are provided for free and created in hard earned free time. If you have constructive feedback we will gladly take it to make the product better but be respectful of others time and effort in doing so. We are not forcing you to use this project.

**Comment by newtonianb** [comment]
@scottkellum I was talking about compass for the defunct part fyi, this project is great did not mean to offend ...

**Comment by Snugug** [comment]
@newtonianb The problem was the aggressive tone clearly upset that we support Compass and channeling that aggressive tone towards the project maintainers for doing something you believe is wrong. The phrase "you catch more flies with honey than vinegar" applies very much so to open source projects. The comment, as a whole, made even more frustrating for us as the most recent comments literally describe how Compass is no longer a dependency we have instructions in our README explaining how to install it without Ruby or Compass.

**Comment by newtonianb** [comment]
Bottom line is no offense was intended, a point of view naturally has a tone to it

Here is my edit

Unfortunately, compass is slowly becoming phased out in medium/large projects due to performance problems. Node Sass has addressed some of the performance concerns and with libraries can almost entirely replace compass. It would be fantastic if we could use this great library without compass. I just discovered this library, it looks great thank you for it.

**Comment by scottkellum** [comment]
Singularity no longer has Compass as a requirement. Please [see the readme](https://github.com/at-import/Singularity#installation) for details.
